dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/EventsAnalyticsQueryCriteria.java

```diff
@@ -221,6 +221,12 @@
     // Event query analytics only
     // -------------------------------------------------------------------------
 
+    /**
+     * Field to base geospatial event analytics on. Default is event. Can be set
+     * to identifiers of attributes and data elements of value type coordinate.
+     */
+    private String fallbackCoordinateField;
```

Comment:
Could we please add validation for this field, to check whether it is within a) a set of predefined values like TEI, ENROLLMENT, ORG_UNIT, etc, or b) a valid UID (see `CodeGenerator#isValidUid`).

In general we are trying to make the analytics APIs "waterproof", so that invalid/unexpected input should return a 409 validation message/error code to avoid confusion/crashes.